### PR TITLE
Bioformats

### PR DESCRIPTION
**Pull request by CGDogan**
The easiest way to test this turns out toı be PNGs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ Containerized IIP
 
 ## building and running
 
+Unless using caMicroscope Distro Docker, [BFBridge](https://github.com/camicroscope/BFBridge) needs to be cloned and placed next to iipsrv/, so that this project's root iipimage/ has subfolders iipimage/ and BFBridge/
```
Comment (birm) [inline README.md:6]: If I'm understanding correctly, you're saying that your options are to either use the dockerfile or add the dependency manually; is that right?

Comment (CGDogan) [reply README.md:6]: Yes

**Comment by birm** [comment]
I've gotten an issue building this:


```
Step 23/26 : RUN make
 ---> Running in 22825bcd8176
Making all in fcgi
make[1]: Entering directory '/root/src/iipsrv/fcgi'
Making all in libfcgi
make[2]: Entering directory '/root/src/iipsrv/fcgi/libfcgi'
gcc -DPACKAGE_NAME=\"iipsrv\" -DPACKAGE_TARNAME=\"iipsrv\" -DPACKAGE_VERSION=\"1.0\" -DPACKAGE_STRING=\"iipsrv\ 1.0\" -DPACKAGE_BUGREPORT=\"ruven@users.sourceforge.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"iipsrv\" -DVERSION=\"1.0\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE__BOOL=1 -DHAVE_STDBOOL_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_GLOB_H=1 -DHAVE_TIME_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_MALLOC=1 -DHAVE_LOG2=1 -DHAVE_SETENV=1 -DHAVE_EXT_POOL_ALLOCATOR_H=1 -DHAVE_UNORDERED_MAP=/\*\*/ -DHAVE_TR1_UNORDERED_MAP=/\*\*/ -DHAVE_NAMESPACES=/\*\*/ -DHAVE_EXT_HASH_MAP=/\*\*/ -DHAVE_ISFINITE=1 -DHAVE_JPEG=1 -DHAVE_LIBMEMCACHED_MEMCACHED_H=1 -DHAVE_MEMCACHED=1 -DHAVE_TIFF=1 -DNPROCS=1 -DHAVE_PNG_H=1 -DHAVE_STREAMBUF_CHAR_TYPE=1 -DHAVE_LIBNSL=1 -DHAVE_FPOS=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_NETDB_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_ARPA_INET_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DHAVE_UNISTD_H=1 -DHAVE_FILENO_PROTO=1 -DHAVE_STRERROR=1 -I. -I. -I../../fcgi/include -I../../fcgi   -g -O2 -Wall -MT libfcgi_a-fcgiapp.o -MD -MP -MF .deps/libfcgi_a-fcgiapp.Tpo -c -o libfcgi_a-fcgiapp.o `test -f 'fcgiapp.c' || echo './'`fcgiapp.c
In file included from fcgiapp.c:28:
../../fcgi/fcgi_config.h:89: warning: "PACKAGE" redefined
   89 | #define PACKAGE "fcgi"
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:92: warning: "PACKAGE_BUGREPORT" redefined
   92 | #define PACKAGE_BUGREPORT ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:95: warning: "PACKAGE_NAME" redefined
   95 | #define PACKAGE_NAME ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:98: warning: "PACKAGE_STRING" redefined
   98 | #define PACKAGE_STRING ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:101: warning: "PACKAGE_TARNAME" redefined
  101 | #define PACKAGE_TARNAME ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:104: warning: "PACKAGE_VERSION" redefined
  104 | #define PACKAGE_VERSION ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:117: warning: "VERSION" redefined
  117 | #define VERSION "2.4.0"
      |
<command-line>: note: this is the location of the previous definition
fcgiapp.c:14:19: warning: 'rcsid' defined but not used [-Wunused-const-variable=]
   14 | static const char rcsid[] = "$Id: fcgiapp.c,v 1.34 2001/12/12 22:54:10 robs Exp $";
      |                   ^~~~~
mv -f .deps/libfcgi_a-fcgiapp.Tpo .deps/libfcgi_a-fcgiapp.Po
gcc -DPACKAGE_NAME=\"iipsrv\" -DPACKAGE_TARNAME=\"iipsrv\" -DPACKAGE_VERSION=\"1.0\" -DPACKAGE_STRING=\"iipsrv\ 1.0\" -DPACKAGE_BUGREPORT=\"ruven@users.sourceforge.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"iipsrv\" -DVERSION=\"1.0\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE__BOOL=1 -DHAVE_STDBOOL_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_GLOB_H=1 -DHAVE_TIME_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_MALLOC=1 -DHAVE_LOG2=1 -DHAVE_SETENV=1 -DHAVE_EXT_POOL_ALLOCATOR_H=1 -DHAVE_UNORDERED_MAP=/\*\*/ -DHAVE_TR1_UNORDERED_MAP=/\*\*/ -DHAVE_NAMESPACES=/\*\*/ -DHAVE_EXT_HASH_MAP=/\*\*/ -DHAVE_ISFINITE=1 -DHAVE_JPEG=1 -DHAVE_LIBMEMCACHED_MEMCACHED_H=1 -DHAVE_MEMCACHED=1 -DHAVE_TIFF=1 -DNPROCS=1 -DHAVE_PNG_H=1 -DHAVE_STREAMBUF_CHAR_TYPE=1 -DHAVE_LIBNSL=1 -DHAVE_FPOS=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_NETDB_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_ARPA_INET_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DHAVE_UNISTD_H=1 -DHAVE_FILENO_PROTO=1 -DHAVE_STRERROR=1 -I. -I. -I../../fcgi/include -I../../fcgi   -g -O2 -Wall -MT libfcgi_a-fcgi_stdio.o -MD -MP -MF .deps/libfcgi_a-fcgi_stdio.Tpo -c -o libfcgi_a-fcgi_stdio.o `test -f 'fcgi_stdio.c' || echo './'`fcgi_stdio.c
In file included from fcgi_stdio.c:23:
../../fcgi/fcgi_config.h:89: warning: "PACKAGE" redefined
   89 | #define PACKAGE "fcgi"
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:92: warning: "PACKAGE_BUGREPORT" redefined
   92 | #define PACKAGE_BUGREPORT ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:95: warning: "PACKAGE_NAME" redefined
   95 | #define PACKAGE_NAME ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:98: warning: "PACKAGE_STRING" redefined
   98 | #define PACKAGE_STRING ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:101: warning: "PACKAGE_TARNAME" redefined
  101 | #define PACKAGE_TARNAME ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:104: warning: "PACKAGE_VERSION" redefined
  104 | #define PACKAGE_VERSION ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:117: warning: "VERSION" redefined
  117 | #define VERSION "2.4.0"
      |
<command-line>: note: this is the location of the previous definition
fcgi_stdio.c:15:19: warning: 'rcsid' defined but not used [-Wunused-const-variable=]
   15 | static const char rcsid[] = "$Id: fcgi_stdio.c,v 1.14 2001/09/01 01:09:30 robs Exp $";
      |                   ^~~~~
mv -f .deps/libfcgi_a-fcgi_stdio.Tpo .deps/libfcgi_a-fcgi_stdio.Po
gcc -DPACKAGE_NAME=\"iipsrv\" -DPACKAGE_TARNAME=\"iipsrv\" -DPACKAGE_VERSION=\"1.0\" -DPACKAGE_STRING=\"iipsrv\ 1.0\" -DPACKAGE_BUGREPORT=\"ruven@users.sourceforge.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"iipsrv\" -DVERSION=\"1.0\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE__BOOL=1 -DHAVE_STDBOOL_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_GLOB_H=1 -DHAVE_TIME_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_MALLOC=1 -DHAVE_LOG2=1 -DHAVE_SETENV=1 -DHAVE_EXT_POOL_ALLOCATOR_H=1 -DHAVE_UNORDERED_MAP=/\*\*/ -DHAVE_TR1_UNORDERED_MAP=/\*\*/ -DHAVE_NAMESPACES=/\*\*/ -DHAVE_EXT_HASH_MAP=/\*\*/ -DHAVE_ISFINITE=1 -DHAVE_JPEG=1 -DHAVE_LIBMEMCACHED_MEMCACHED_H=1 -DHAVE_MEMCACHED=1 -DHAVE_TIFF=1 -DNPROCS=1 -DHAVE_PNG_H=1 -DHAVE_STREAMBUF_CHAR_TYPE=1 -DHAVE_LIBNSL=1 -DHAVE_FPOS=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_NETDB_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_ARPA_INET_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DHAVE_UNISTD_H=1 -DHAVE_FILENO_PROTO=1 -DHAVE_STRERROR=1 -I. -I. -I../../fcgi/include -I../../fcgi   -g -O2 -Wall -MT libfcgi_a-os_unix.o -MD -MP -MF .deps/libfcgi_a-os_unix.Tpo -c -o libfcgi_a-os_unix.o `test -f 'os_unix.c' || echo './'`os_unix.c
In file included from os_unix.c:23:
../../fcgi/fcgi_config.h:89: warning: "PACKAGE" redefined
   89 | #define PACKAGE "fcgi"
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:92: warning: "PACKAGE_BUGREPORT" redefined
   92 | #define PACKAGE_BUGREPORT ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:95: warning: "PACKAGE_NAME" redefined
   95 | #define PACKAGE_NAME ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:98: warning: "PACKAGE_STRING" redefined
   98 | #define PACKAGE_STRING ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:101: warning: "PACKAGE_TARNAME" redefined
  101 | #define PACKAGE_TARNAME ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:104: warning: "PACKAGE_VERSION" redefined
  104 | #define PACKAGE_VERSION ""
      |
<command-line>: note: this is the location of the previous definition
../../fcgi/fcgi_config.h:117: warning: "VERSION" redefined
  117 | #define VERSION "2.4.0"
      |
<command-line>: note: this is the location of the previous definition
os_unix.c: In function 'OS_Accept':
os_unix.c:1166:70: warning: pointer targets in passing argument 3 of 'accept' differ in signedness [-Wpointer-sign]
 1166 |                 socket = accept(listen_sock, (struct sockaddr *)&sa, &len);
      |                                                                      ^~~~
      |                                                                      |
      |                                                                      int *
In file included from /usr/include/netinet/in.h:23,
                 from os_unix.c:28:
/usr/include/x86_64-linux-gnu/sys/socket.h:307:42: note: expected 'socklen_t * restrict' {aka 'unsigned int * restrict'} but argument is of type 'int *'
  307 |                    socklen_t *__restrict __addr_len);
      |                    ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
os_unix.c: In function 'OS_IsFcgi':
os_unix.c:1266:51: warning: pointer targets in passing argument 3 of 'getpeername' differ in signedness [-Wpointer-sign]
 1266 |     if (getpeername(sock, (struct sockaddr *)&sa, &len) != 0 && errno == ENOTCONN) {
      |                                                   ^~~~
      |                                                   |
      |                                                   int *
/usr/include/x86_64-linux-gnu/sys/socket.h:131:47: note: expected 'socklen_t * restrict' {aka 'unsigned int * restrict'} but argument is of type 'int *'
  131 |                         socklen_t *__restrict __len) __THROW;
      |                         ~~~~~~~~~~~~~~~~~~~~~~^~~~~
os_unix.c: At top level:
os_unix.c:20:19: warning: 'rcsid' defined but not used [-Wunused-const-variable=]
   20 | static const char rcsid[] = "$Id: os_unix.c,v 1.37 2002/03/05 19:14:49 robs Exp $";
      |                   ^~~~~
mv -f .deps/libfcgi_a-os_unix.Tpo .deps/libfcgi_a-os_unix.Po
rm -f libfcgi.a
ar cru libfcgi.a  libfcgi_a-fcgiapp.o libfcgi_a-fcgi_stdio.o libfcgi_a-os_unix.o
ar: `u' modifier ignored since `D' is the default (see `U')
ranlib libfcgi.a
make[2]: Leaving directory '/root/src/iipsrv/fcgi/libfcgi'
Making all in include
make[2]: Entering directory '/root/src/iipsrv/fcgi/include'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/root/src/iipsrv/fcgi/include'
make[2]: Entering directory '/root/src/iipsrv/fcgi'
make[2]: Nothing to be done for 'all-am'.
make[2]: Leaving directory '/root/src/iipsrv/fcgi'
make[1]: Leaving directory '/root/src/iipsrv/fcgi'
Making all in src
make[1]: Entering directory '/root/src/iipsrv/src'
g++ -DPACKAGE_NAME=\"iipsrv\" -DPACKAGE_TARNAME=\"iipsrv\" -DPACKAGE_VERSION=\"1.0\" -DPACKAGE_STRING=\"iipsrv\ 1.0\" -DPACKAGE_BUGREPORT=\"ruven@users.sourceforge.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"iipsrv\" -DVERSION=\"1.0\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE__BOOL=1 -DHAVE_STDBOOL_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_GLOB_H=1 -DHAVE_TIME_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_MALLOC=1 -DHAVE_LOG2=1 -DHAVE_SETENV=1 -DHAVE_EXT_POOL_ALLOCATOR_H=1 -DHAVE_UNORDERED_MAP=/\*\*/ -DHAVE_TR1_UNORDERED_MAP=/\*\*/ -DHAVE_NAMESPACES=/\*\*/ -DHAVE_EXT_HASH_MAP=/\*\*/ -DHAVE_ISFINITE=1 -DHAVE_JPEG=1 -DHAVE_LIBMEMCACHED_MEMCACHED_H=1 -DHAVE_MEMCACHED=1 -DHAVE_TIFF=1 -DNPROCS=1 -DHAVE_PNG_H=1 -DHAVE_STREAMBUF_CHAR_TYPE=1 -DHAVE_LIBNSL=1 -DHAVE_FPOS=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_NETDB_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_ARPA_INET_H=1 -DHAVE_SYS_TIME_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DHAVE_UNISTD_H=1 -DHAVE_FILENO_PROTO=1 -DHAVE_STRERROR=1 -I. -I. -I../fcgi/include -I../fcgi   -I/usr/local/include/openslide -DBFBRIDGE_INLINE -I/java_home/include -I/java_home/include/linux -I/java_home/include/darwin -I/java_home/include/win32 -I/java_home/include/bsd  -std=gnu++0x -g -O2 -MT IIPImage.o -MD -MP -MF .deps/IIPImage.Tpo -c -o IIPImage.o IIPImage.cc
In file included from ../../BFBridge/c/bfbridge_basiclib.c:6,
                 from ../../BFBridge/c/bfbridge_basiclib.h:401,
                 from BioFormatsThread.h:19,
                 from BioFormatsInstance.h:15,
                 from BioFormatsManager.h:9,
                 from IIPImage.cc:44:
../../BFBridge/c/bfbridge_basiclib.h:414:2: error: #error "bfbridge_basiclib.h was already "#include"d in non-header-only mode"
  414 | #error "bfbridge_basiclib.h was already "#include"d in non-header-only mode"
      |  ^~~~~
In file included from IIPImage.cc:24:
IIPImage.h:81:24: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
   81 |   void testImageType() throw( file_error );
      |                        ^~~~~
IIPImage.h:239:56: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
  239 |   static time_t getFileTimestamp(const std::string& s) throw( file_error );
      |                                                        ^~~~~
IIPImage.h:246:48: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
  246 |   bool updateTimestamp( const std::string& s ) throw( file_error );
      |                                                ^~~~~
IIPImage.cc:84:32: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
   84 | void IIPImage::testImageType() throw(file_error)
      |                                ^~~~~
IIPImage.cc:427:55: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
  427 | time_t IIPImage::getFileTimestamp(const string& path) throw(file_error)
      |                                                       ^~~~~
IIPImage.cc:440:54: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
  440 | bool IIPImage::updateTimestamp( const string& path ) throw(file_error)
      |                                                      ^~~~~
make[1]: *** [Makefile:515: IIPImage.o] Error 1
make[1]: Leaving directory '/root/src/iipsrv/src'
make: *** [Makefile:405: all-recursive] Error 1
ERROR: Service 'iip' failed to build: The command '/bin/sh -c make' returned a non-zero code: 2

**Comment by CGDogan** [comment]
@birm I forgot to mention that image-decoders needs to be rebuilt to Docker first please.

**Comment by birm** [comment]
I think I've had bad luck with the images from this website (https://downloads.openmicroscopy.org/images/).

Do you have recommendations? I got loader logs like:

``` 
ca-load   | 19:08:31.746 [main] DEBUG loci.common.services.ServiceFactory - CLASSPATH missing interface: ome.codecs.services.LuraWaveService
ca-load   | java.lang.ClassNotFoundException: ome.codecs.services.LuraWaveService
ca-load   | 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
ca-load   | 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
ca-load   | 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
ca-load   | 	at java.base/java.lang.Class.forName0(Native Method)
ca-load   | 	at java.base/java.lang.Class.forName(Class.java:496)
ca-load   | 	at java.base/java.lang.Class.forName(Class.java:475)
ca-load   | 	at loci.common.services.ServiceFactory.init(ServiceFactory.java:157)
ca-load   | 	at loci.common.services.ServiceFactory.services(ServiceFactory.java:249)
ca-load   | 	at loci.common.services.ServiceFactory.<init>(ServiceFactory.java:105)
ca-load   | 	at loci.common.services.ServiceFactory.<init>(ServiceFactory.java:92)
ca-load   | 	at loci.formats.Memoizer.getService(Memoizer.java:798)
ca-load   | 	at loci.formats.Memoizer.setId(Memoizer.java:723)
ca-load   | 	at org.camicroscope.BFBridge.BFOpen(BFBridge.java:218)
ca-load   | 19:08:31.784 [main] DEBUG loci.common.NIOByteBufferProvider - Using mapped byte buffer? false
ca-load   | 19:08:31.874 [main] INFO loci.formats.ImageReader - LIFReader initializing /images/20191025_Test_FRET_585._423_426.lif
ca-load   | 19:08:31.875 [main] DEBUG loci.formats.FormatHandler - LIFReader initializing /images/20191025_Test_FRET_585._423_426.lif
ca-load   | 19:08:31.875 [main] DEBUG loci.formats.FormatHandler - loci.formats.in.LIFReader.initFile(/images/20191025_Test_FRET_585._423_426.lif)
ca-load   | 19:08:31.892 [main] INFO loci.formats.FormatHandler - Reading header
ca-load   | 19:08:32.115 [main] INFO loci.formats.FormatHandler - Finding image offsets
ca-load   | 19:08:32.116 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 553953; 0 blocks read
ca-load   | 19:08:32.117 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 553999; 0 blocks read
ca-load   | 19:08:32.126 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 2651197; 1 blocks read
ca-load   | 19:08:32.136 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 4748395; 2 blocks read
ca-load   | 19:08:32.136 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 4748441; 2 blocks read
ca-load   | 19:08:32.145 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 6845639; 3 blocks read
ca-load   | 19:08:32.153 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 17331445; 4 blocks read
ca-load   | 19:08:32.162 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 19428643; 5 blocks read
ca-load   | 19:08:32.170 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 20477265; 6 blocks read
ca-load   | 19:08:32.179 [main] DEBUG loci.formats.FormatHandler - Looking for a block at 28865919; 7 blocks read
ca-load   | [2023-08-28 19:08:32 +0000] [7] [ERROR] Worker (pid:98) was sent SIGILL!
```

**Comment by CGDogan** [comment]
@birm Running Intel image-decoders on M1 will lead to SIGILL without any warnings before it. It'll also be very slow for you to build. The text before it is actually usual debug logs unfortunately which I couldn't stop and they're being printed all the time.

Unfortunately few files have 1, 3 or 4 channels. The file you tried has two channels so it won't load.

Here's a png file https://commons.wikimedia.org/wiki/File:Transparency_example.PNG

Another file but needs to be unzipped https://drive.google.com/file/d/1set5t3hTIfai6MxSWaQSxjft5C-mMm82/view?usp=sharing

**Comment by CGDogan** [comment]
You can use my image-decoders for native ARM testing `cgd30/image-decoders:latest`

**Review by birm — APPROVED** [review]
Sorry for the delay, I just decided to wait until a WFH day when I could use my usual build system.

**Pull request merged**